### PR TITLE
Feature/enforce signature

### PR DESCRIPTION
**Pull request by ArlonTM**
* Enforce the use of signature
* Deprecate all sec_key-related operations/calls
* Document code
* Update README accordingly
* Perform code cleanup

### CODE REVIEW

**Review by JNdhlovu — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -1,17 +1,46 @@
...
+For usage questions, the best resource is [our official documentation](docs.smileidentity.com). However, if you require further assistance, you can file a [support ticket via our portal](https://portal.smileidentity.com/partner/support/tickets) or visit the [contact us page](https://portal.smileidentity.com/partner/support/tickets) on our website.
 
-Edit the example_core.php file and replace sections marked with <>
+## Contributing
 
-### Run tests
+Bug reports and pull requests are welcome on GitHub at https://github.com/smileidentity/smile-identity-core-php
 
-In same directory run `vendor/bin/phpunit tests`
+Please format the code with [black](https://github.com/psf/black) prior to submitting pull requests, by running `black .` from the project's root.
```
Comment (macphilips) [inline README.md:46]: I didn't realize black also formats PHP code

File: lib/IdApi.php [hunk 2]
```diff
@@ -54,30 +56,26 @@ public function setClient(Client $client): void
...
+        $sec_params = $this->sig_class->generate_signature();
 
         $data = array(
             'language' => 'php',
             'callback_url' => $this->default_callback,
             'partner_params' => $partner_params,
-            'partner_id' => $this->partner_id
+            'partner_id' => $this->partner_id,
+            'source_sdk' => 'PHP',
+            'source_sdk_version' => '2.0.0'
```
Comment (macphilips) [inline lib/IdApi.php:78]: The latest release of the lib points to `3.0.0`

Comment (macphilips) [reply lib/IdApi.php:78]: Also, can we use `Config::VERSION` instead of hardcoding the value

File: README.md (hunk 1 above)
Comment (ArlonTM) [reply README.md:46]: Don't be smart with me now @macphilips 😂😂😂
Unfortunate copy-paste! 🙏🏽

File: tests/SmileIdentityCoreTest.php [hunk 3]
```diff
@@ -343,11 +334,11 @@ public function testSubmitJobShouldRequiresAtLeastOneSelfieForJTOtherThanJT6():
         $this->sic->submit_job($this->partnerParams, $imageDetails, $this->idParams, $this->options);
     }
 
-    public function testGenerateKey(): void
+    public function testGenerateSignature(): void
     {
-        $timestamp = Clock::now()->getTimestamp();
-        $sec_key = $this->sic->generate_sec_key(false);
-        $this->assertEquals($timestamp, $sec_key["timestamp"]);
+        $timestamp = Clock::now()->format(DateTimeInterface::ISO8601);
```
Comment (macphilips) [inline tests/SmileIdentityCoreTest.php:339]: `DateTimeInterface::ISO8601` has been deprecated. can you update to `DateTimeInterface::ATOM`
```suggestion
        $timestamp = Clock::now()->format(DateTimeInterface::ATOM);
```

**Review by macphilips — APPROVED** [review]
LGTM

**Review by JNdhlovu — APPROVED** [review]

**Pull request merged**